management/src/main/java/org/ehcache/management/registry/DefaultCollectorService.java

```diff
@@ -217,8 +217,23 @@ public synchronized void stopStatisticCollector() {
 
   @Override
   public void updateCollectedStatistics(String capabilityName, Collection<String> statisticNames) {
-    StatisticQuery.Builder builder = managementRegistry.withCapability(capabilityName).queryStatistics(statisticNames);
-    selectedStatsPerCapability.put(capabilityName, builder);
+    if(!statisticNames.isEmpty()) {
+      StatisticQuery.Builder builder = managementRegistry.withCapability(capabilityName).queryStatistics(statisticNames);
+      selectedStatsPerCapability.put(capabilityName, builder);
+    } else {
+      // we clear the stats set
+      selectedStatsPerCapability.remove(capabilityName);
+    }
+  }
+
+  // for test purposes
+  ConcurrentMap<String, StatisticQuery.Builder> getSelectedStatsPerCapability() {
+    return selectedStatsPerCapability;
```

Comment:
Even for tests, I would make sure this is not modifiable. And thus only the Map interface is needed.